### DISCUSSION TITLE
Ale and `elixir-tools`

### DISCUSSION THREAD

**Discussion by Bastes** [w0rp/ale#4878] [Q&A - Ask for help with problems]
Hello,

I'm not sure I'm in the right section but as a happy ALE user for most of my lsp/linting uses, I've found `elixir-ls` wanting when working on different versions of Elixir, as it downloads and compiles a one-size-fits-all version of `elixir-ls` that may not be compatible with every elixir version installed. For example, I'm currently working with my team on an elixir `v1.11.4` which plays well with the version of `elixir-ls` installed (`0.12`, ironically the last one supporting elixir `v1.11`), but I've got other projects on `v1.17` which is not playing well with `elixir-ls` `v0.12.0`.

I've since toyed with [elixir-tools](https://github.com/elixir-tools/elixir-tools.nvim) which comes with another policy: compiling `elixir-ls` in the relative `.elixir_ls` subdirectory of a project so as to pick a version that works with the version of elixir of that project. This seems the way to go for someone working with multiple elixir versions.

So, here's my question: is there a way to tell ALE to go searching for a `language_server.sh` relative to the project? So far I tried to give a relative path to ALE and it wasn't able to start the server so I thought I'd ask, and if that is not possible, maybe suggest this feature? I might even help if that's not above my abilities.

Anyways, great plugin :) always using it.

**Comment by hsanson** [comment]
There does not seem to be a way to do this now. Modifying the `ale_linters#elixir#elixir_ls#GetExecutable()` function to search first for the executable within the project root .elixir_ls path and then fallback to the global one would do. There is a helper function that could be used for this `ale#path#FindNearestExecutable()`.

**Reply by Bastes** [reply]
Thank you for your reply :) I'll be looking into it, I'm not a very strong vim scripter but if I can I'll be testing and ideally providing a PR.
